where-to-travel/src/main/webapp/script.js

```diff
@@ -226,59 +339,131 @@ function addPlacesFromDirection(lat, lng, place_candidates) {
-    for (var i = 0; i < 25; i++) {
-      const lat = listPlaces[i].geometry.location.lat();
-      const lng = listPlaces[i].geometry.location.lng();
+    let destinations = [];
+    
+    // Iterate through places to get all latitudes and longitudes of destinations 
+    for (place of places) {
+      let lat = place.geometry.location.lat();
+      let lng = place.geometry.location.lng();
       let destination = new google.maps.LatLng(lat, lng);
-      userDestinations.push(destination);
+      destinations.push(destination);
     }
 
-    var service = new google.maps.DistanceMatrixService();
+    let service = new google.maps.DistanceMatrixService();
     service.getDistanceMatrix({
-      origins: [userLocation],
-      destinations: userDestinations,
+      origins: [home],
+      destinations: destinations,
       travelMode: 'DRIVING',
       unitSystem: google.maps.UnitSystem.IMPERIAL,
     }, callback);
 
     function callback(response, status) {    
+      let total_time = 0;
+      let total_places = 0;
+
       if (status == 'OK') {
-        var origins = response.originAddresses;
-        var destinations = response.destinationAddresses;
-      
-        for (var i = 0; i < origins.length; i++) {
-          var results = response.rows[i].elements;
-          for (var j = 0; j < results.length; j++) {
-            var element = results[j];
-            //Check if the time is within the +- 30 min = 1800 sec range
-            if (element.duration.value < time + 1800 && element.duration.value > time - 1800) {
-              acceptablePlaces.push({
-                "name": listPlaces[j].name,
-                "geometry" : listPlaces[j].geometry,
-                "timeInSeconds": element.duration.value,
-                "timeAsString": element.duration.text
+        // There is only one origin
+        let results = response.rows[0].elements;
+        const ThirtyMinsInSecs = 1800;
+
+        for (let j = 0; j < results.length; j++) {
+          let destination_info = results[j];
+
+          if (destination_info.status == 'OK') {
+            let destination_time = destination_info.duration.value; 
+         
+            total_time += destination_time;
+            total_places += 1;
+
+            // Check if the destination time is within +- 30 minutes of requested travel time
+            if (destination_time <= time + ThirtyMinsInSecs && destination_time >= time - ThirtyMinsInSecs) {
+              acceptablePlacesInfo.places.push({
+                name: places[j].name,
+                geometry: places[j].geometry,
+                timeInSeconds: destination_time,
+                timeAsString: destination_info.duration.text
               });
             }
           }
         }
       }
-      resolve(acceptablePlaces);
-    }
+      
+      if (total_places == 0) {
+        acceptablePlacesInfo.avg_time = 0;
+      } else {
+        acceptablePlacesInfo.avg_time = total_time/total_places;
```

Comment:
You're right. I have moved the calculation for the average time to outside of this function once all of the filter results are aggregated. I would still like to keep the check for total_places being 0 because this may possibly happen if the bounding box is on some island that is not accessible by road from the origin and in that case, none of the places in that bounding box would have a valid return status from the distance matrix api and be processed.